### PR TITLE
Added engraving and notation project concept

### PR DESCRIPTION
**Pull request by igorkorsukov**
In addition to the score itself, a mscz file also stores other data, for example, settings for a synthesizer, mixer, omr, etc.
So, we should talk not just about a score, but about a Project.
To some extent, the project functions are performed by MasterScore, but this is not its responsibility, this functionality only overloads it and confuses what, where and why.

I add explicitly the concept of a project.

### CODE REVIEW

File: src/importexport/guitarpro/internal/importgtp.cpp [hunk 1]
```diff
@@ -2975,7 +2975,7 @@ Score::FileError importGTP(MasterScore* score, const QString& name)
             continue;
         }
         QMultiMap<int, int> tracks;
-        Score* pscore = new Score(score);
+        Score* pscore =  score->createScore();
```
Comment (worldwideweary) [inline src/importexport/guitarpro/internal/importgtp.cpp:2978]: Extra space? tsk tsk!

Comment (igorkorsukov) [reply src/importexport/guitarpro/internal/importgtp.cpp]: thanks, fixed

File: src/notation/internal/notationproject.cpp [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    Ret ret = reader->read(score, path, options);
+    if (!ret) {
+        return ret;
+    }
+
+    if (!Ms::MScore::lastError.isEmpty()) {
+        LOGE() << Ms::MScore::lastError;
+    }
+
+    project->setupMasterScore();
```
Comment (igorkorsukov) [inline src/notation/internal/notationproject.cpp:84]: In general, the loading process is ugly right now, I will try to make it better in the next PR

**Review by Eism — APPROVED** [review]

File: src/userscores/view/templatepaintview.cpp [hunk 3]
```diff
@@ -63,16 +63,16 @@ QString TemplatePaintView::zoomOutSequence() const
...
 
     if (!ret) {
         LOGE() << ret.toString();
     }
 
-    setNotation(masterNotation->notation());
+    setNotation(notationProject->masterNotation()->notation());
 
-    if (masterNotation) {
+    if (notationProject) {
```
Comment (wizofaus) [inline src/userscores/view/templatepaintview.cpp:75]: Something seems wrong here (though it seemed wrong already), the notation project gets deleted when the shared_ptr goes out of scope, but there are still references to it (via the setNotation call), which is causing MuseScore to crash doing almost anything now. @igorkorsukov

**Pull request merged**